### PR TITLE
fix(#9963):Add logic to handle for f.Close() for util/io/files/tar.go

### PR DESCRIPTION
**Pull request by my-git9**
Add logic to handle for f.Close() for util/io/files/tar.go

part of issue: https://github.com/argoproj/argo-cd/issues/9963

Checklist:
* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by my-git9** [comment]
@crenshaw-dev Hi, I can't fix the fail check, can you help me?

File: util/io/files/tar.go [hunk 1]
```diff
@@ -177,7 +177,12 @@ func (t *tgz) tgzFile(path string, fi os.FileInfo, err error) error {
 		if err != nil {
 			return fmt.Errorf("error opening file %q: %w", fi.Name(), err)
 		}
-		defer f.Close()
+		defer func() {
+			err := f.Close()
+			if err != nil {
+				panic(fmt.Sprintf("error closing file %q: %v", fi.Name(), err))
```
Comment (crenshaw-dev) [inline util/io/files/tar.go:183]: I think in this case we can just log the error rather than panicking.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: util/io/files/tar.go (hunk 1 above)
Comment (my-git9) [reply util/io/files/tar.go]: Thanks, modified, is this ok?

Comment (crenshaw-dev) [reply util/io/files/tar.go]: Yes, thank you!

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**